### PR TITLE
[WIP][CARBONDATA-3769]Upgrade hadoop versionto 3.1.1 and add profile for 2.7.2

### PR DESCRIPTION
**Pull request by akashrn5**
### Why is this PR needed?
 since we have removed old spark version, Still we are using adoop 2.7.2 as default.
 
 ### What changes were proposed in this PR?
Upgrade to hadoop 3.1.1 and add maven profiles for old hadoop versions
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - No(existing tests will take care)

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/981/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2694/

File: processing/pom.xml [hunk 1]
```diff
@@ -75,6 +75,11 @@
       <artifactId>junit</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
```
Comment (QiangCai) [inline processing/pom.xml:78]: can you add dependency into profile hadoop-3.1

Comment (akashrn5) [reply processing/pom.xml:78]: currently there is profile for hadoop, do you want me to add a profile for it? As it was default i didnt add

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/982/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2695/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/984/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2697/

**Comment by akashrn5** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/985/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2698/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/986/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2699/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/987/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2700/

File: integration/hive/src/test/java/org/apache/carbondata/hive/HiveCarbonTest.java [hunk 2]
```diff
@@ -65,7 +65,11 @@ public static void setup() throws Exception {
   public void verifyDataAfterLoad() throws Exception {
     statement.execute("drop table if exists hive_carbon_table4");
     statement.execute("CREATE TABLE hive_carbon_table4(shortField SMALLINT , intField INT, bigintField BIGINT , doubleField DOUBLE, stringField STRING, timestampField TIMESTAMP, decimalField DECIMAL(18,2), dateField DATE, charField CHAR(5), floatField FLOAT) stored by 'org.apache.carbondata.hive.CarbonStorageHandler'");
-    statement.execute("insert into hive_carbon_table4 select * from hive_table");
+    try {
+      statement.execute("insert into hive_carbon_table4 select * from hive_table");
+    } catch (Exception ex) {
```
Comment (Indhumathi27) [inline integration/hive/src/test/java/org/apache/carbondata/hive/HiveCarbonTest.java:70]: why insert will fail?

Comment (akashrn5) [reply integration/hive/src/test/java/org/apache/carbondata/hive/HiveCarbonTest.java:70]: there are some issues, im just adding for my reference, not actual code

Comment (Indhumathi27) [reply integration/hive/src/test/java/org/apache/carbondata/hive/HiveCarbonTest.java:70]: okay

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/990/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2703/

**Pull request closed without merging**